pennylane/tape/qscript.py

```diff
@@ -356,6 +359,21 @@ def diagonalizing_gates(self) -> List[Operation]:
                     rotation_gates.extend(observable.diagonalizing_gates())
         return rotation_gates
 
+    @property
+    def shots(self):
+        """Returns a ``Shots`` object containing information about the number
+        and batches of shots
+
+        Returns:
+            ~.Shots: Object with shot information
+        """
+        return self._shots
+
+    @shots.setter
+    def shots(self, shots):
+        """Setter for shots information stored on the ``QuantumScript``."""
+        self._shots = Shots(shots)
+
```

Comment:
```suggestion
```

Let's get rid of the setter for now to discourage the use of dynamically modifying attributes.  If we come across a use case where we end up really needing this ability, we can reconsider this decision then, but its easier to add in behavior then take it away.